hazelcast/src/test/java/com/hazelcast/internal/blackbox/impl/MethodSensorInputTest.java

```diff
@@ -0,0 +1,280 @@
+        SomeSource source = new SomeSource();
+        Method method = source.getClass().getDeclaredMethod(fieldName);
+        SensorInput sensorInput = method.getAnnotation(SensorInput.class);
+        MethodSensorInput input = new MethodSensorInput(method, sensorInput);
+        assertFalse(input.isDouble());
+
+        long value = input.getLong(source);
+
+        assertEquals(expectedValue, value);
+    }
+
+    @Test
+    public void getLong_whenDouble() throws Exception {
+        getLong_whenDouble("floatMethod");
+        getLong_whenDouble("doubleMethod");
+    }
+
+    public void getLong_whenDouble(String fieldName) throws Exception {
+        SomeSource source = new SomeSource();
+        Method method = source.getClass().getDeclaredMethod(fieldName);
+        SensorInput sensorInput = method.getAnnotation(SensorInput.class);
+        MethodSensorInput fieldSensorInput = new MethodSensorInput(method, sensorInput);
+
+        try {
+            fieldSensorInput.getLong(source);
+            fail();
+        } catch (IllegalStateException expected) {
+
+        }
+    }
+
+    @Test
+    public void getDouble() throws Exception {
+        getDouble("floatMethod", 10);
+        getDouble("doubleMethod", 10);
+        getDouble("DoubleMethod", 10);
+        getDouble("FloatMethod", 10);
+        getDouble("nullDoubleMethod", 0);
+        getDouble("nullFloatMethod", 0);
+    }
+
+    public void getDouble(String fieldName, double expected) throws Exception {
+        SomeSource source = new SomeSource();
+        Method method = source.getClass().getDeclaredMethod(fieldName);
+        SensorInput sensorInput = method.getAnnotation(SensorInput.class);
+
+        MethodSensorInput fieldSensorInput = new MethodSensorInput(method, sensorInput);
+        assertTrue(fieldSensorInput.isDouble());
+
+        double value = fieldSensorInput.getDouble(source);
+
+        assertEquals(expected, value, 0.1);
+    }
+
+    @Test
+    public void getDouble_whenLong() throws Exception {
+        getDouble_whenLong("byteMethod");
+        getDouble_whenLong("shortMethod");
+        getDouble_whenLong("intMethod");
+        getDouble_whenLong("longMethod");
+        getDouble_whenLong("atomicLongMethod");
+        getDouble_whenLong("atomicIntegerMethod");
+        getDouble_whenLong("counterMethod");
+        getDouble_whenLong("collectionMethod");
+    }
+
+    public void getDouble_whenLong(String fieldName) throws Exception {
+        SomeSource source = new SomeSource();
+        Method field = source.getClass().getDeclaredMethod(fieldName);
+        SensorInput sensorInput = field.getAnnotation(SensorInput.class);
+        MethodSensorInput fieldSensorInput = new MethodSensorInput(field, sensorInput);
+
+        try {
+            fieldSensorInput.getDouble(source);
+            fail();
+        } catch (IllegalStateException expected) {
+
+        }
+    }
+
+    private class SomeSource {
```

Comment:
just a tip: testing extreme values such as `Integer.MAX_VALUE` and `Integer.MIN_VALUE` would make it a better test.